src/components/views/dialogs/InviteDialog.tsx

```diff
@@ -684,7 +684,8 @@ export default class InviteDialog extends React.PureComponent<Props, IInviteDial
                                 display_name: profile.displayname,
                                 avatar_url: profile.avatar_url,
                             }),
-                            userId: lookup.mxid,
+                            // Use the search term as identifier, so that it shows up in suggestions.
+                            userId: term,
```

Comment:
This is brittle and confusing, this isn't a userId (read MXID). In all other cases `userId` maps to `user.X` where X is the key which holds the MXID. Feels like you should add a `filter_term` field to `threepidResultsMixin` results which can be used to hold the original query or email or whatever you like more generically. Or maybe a `is_match=true` or similar.

Reply:
It is already used this way in other places  :wink:  https://github.com/matrix-org/matrix-react-sdk/blob/develop/src/components/views/dialogs/InviteDialog.tsx#L654-L657

I can try to change it anyway.

Reply:
Ah yes, did miss that incantation. My bad.

Reply:
I think I prefer your simpler first solution :P